sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1391,11 +1391,30 @@ class Analyzer(
               notMatchedActions = newNotMatchedActions)
         }
 
+      case f @ Filter(cond, _) if containsAggregate(cond) => f
       case q: LogicalPlan =>
         logTrace(s"Attempting to resolve ${q.simpleString(SQLConf.get.maxToStringFields)}")
+        //        q.mapExpressions { e =>
+        //          q match {
+        //            case _: Filter if containsAggregate(e) =>
+        //              e
+        //            case _ =>
+        //              resolveExpressionTopDown(e, q)
+        //          }
+        //        }
         q.mapExpressions(resolveExpressionTopDown(_, q))
     }
 
+    def containsAggregate(e: Expression): Boolean = {
```

Comment:
why can't we reuse `ResolveAggregateFunctions.containsAggregate`?

Reply:
> why can't we reuse `ResolveAggregateFunctions.containsAggregate`?

Since here function is still UnresolvedFunction, we can't just reuse this.